### ISSUE TITLE
Allow same-line braces for empty constructors

### ISSUE THREAD

**Issue by Crell** [FriendsOfPHP/PHP-CS-Fixer#5632]
## Bug report

Coming over here from https://github.com/nunomaduro/phpinsights/issues/475.

PHP 8.0, PHP CS 3.6.0, run through phpinsights.

### Code snippet that reproduces the problem

I'm not sure if this is more feature request or bug.  Refile as needed.

With constructor promotion in PHP 8, it's now common to have a constructor with no body.  The most natural way to write that is this:

```php
    public function __construct(
        public ?int $priority,
        public ?string $id = null,
        public ?string $type = null,
    ) {}
```

However, the `PhpCsFixer\Fixer\Basic\BracesFixer` check complains about the curly braces being on the same line.  It wants you to write it like this instead:

```php
    public function __construct(
        public ?int $priority,
        public ?string $id = null,
        public ?string $type = null,
    ) {
    }
```

And... no, I will not say that is an improvement no matter how much you pay me. :smile: 

Can the braces check detect that an empty set of braces does not, in fact, need to be split to multiple lines?  (This likely applies to empty exceptions, too, but I've not hit that yet myself.)

**Comment by kubawerlos** [comment]
> PHP CS 3.6.0

Would be more useful if you share version of PHP CS Fixer 😉 

> I will not say that is an improvement no matter how much you pay me.

Can you instead say why it is bug? It looks like not explained in any PSR and "most natural way to write" is just your opinion (personally I agree, but it doesn't matter) and someone else can say "breaking line is better for consistency".

Moreover, I'm not sure if anyone want to touch BracesFixers before this monster is merged: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4884.

**Comment by Crell** [comment]
> Would be more useful if you share version of PHP CS Fixer 

Bah.  phpcs, php-cs-fixer, the binaries are too close...  :stuck_out_tongue: Updated the description.

Merging that beast first makes total sense, yes.  This isn't a rush.

As for why, well, all whitespace is subjective.  I would argue that sigils that are in essence vestigial noise (as in this case) are best left inline, to minimize vertical noise.  The more compact version I recommend offers "one line of intro, list of args, one line of closing stuff."  The current suggested format is "one line of intro, list of args, two lines of closing stuff."  Why the asymmetry?  I'd argue the asymmetry is undesireable. and harmful for readability.

**Comment by keradus** [comment]
https://www.php-fig.org/psr/psr-12/#44-methods-and-functions
> Method and function names MUST NOT be declared with space after the method name. **The opening brace MUST go on its own line**, and the closing brace MUST go on the next line following the body.

The rule works according to the PSR standard it aims to implement.

If you want to change the behaviour that is designed, it's a new feature that can be implemented only behind a configuration flag.
I'm OK to welcome decent PR with nice code review for this request (yet not interested myself to provide it)

**Comment by francescolaffi** [comment]
FYI i was searching for this and actually found 2 issues, with https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5475 preceding this one but with less conversation, in case you'd like close one.

**Comment by julienfalque** [comment]
Indeed, this is a duplicate of #5475, let's continue the discussion there :)

**Comment by raveren** [comment]
It's a long purple link road to get to the answer from this google result, but this was implemented and if you want to use it, add the rule `'single_line_empty_body'` to your config, example:

```php
// as close to PER as possible 
return (new PhpCsFixer\Config())->setRules([
    '@PSR12'                 => true,
    'function_declaration'   => ['closure_function_spacing' => 'none', 'closure_fn_spacing' => 'none'],
    'no_break_comment'       => false,
    'single_line_empty_body' => true, // this!
])
    ->setFinder($finder)
```
